### PR TITLE
Added getting started doc for Raspberry PI.

### PR DESCRIPTION
**Pull request by InconceivableDuck**

### CODE REVIEW

File: docs/getting-started/boards/getting-started-with-raspberry-pi.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
+# Getting Started with the Raspberry PI
+
+The <a href="https://www.raspberrypi.org/" target="_blank">Raspberry PI</a> is a very popular compute module commonly used in IoT related projects. This guide covers how to connect the Raspberry PI to the Structure Platform.
+
+The example for this guide is a button that is connected to the GPIO on the Raspberry PI. Whenever the button is pressed, a [state](/devices/state) message will be sent to the platform that can be used to trigger actions. To demonstrate device [commands](/devices/state), there is also an LED connected to GPIO that the platform can turn on and off.
+
+The source code for this example is <a href="https://github.com/GetStructure/example-raspberry-pi" target="_blank">available on GitHub</a>.
+
+## Install Node.js on the Raspberry PI
```
Comment (cephirothdy2j) [inline docs/getting-started/boards/getting-started-with-raspberry-pi.md]: This guide makes no mention of installing Raspbian; the blog entry on installing Node "assumes you're using the Raspian OS", which may be enough.

Maybe we should start with a "What You'll Need" section and include everything necessary to get through this tutorial: a Pi, USB keyboard, jumper wires, button, etc.

File: docs/getting-started/boards/getting-started-with-raspberry-pi.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+We're going to use the <a href="https://github.com/GetStructure/structure-api-js" target="_blank">Structure JavaScript SDK</a> to connect the Raspberry PI to the Structure Platform. Follow the instructions <a href="https://www.getstructure.io/blog/how-to-install-nodejs-on-raspberry-pi" target="_blank">on our blog</a> for how to install Node.js.
+
```
Comment (cephirothdy2j) [inline docs/getting-started/boards/getting-started-with-raspberry-pi.md]: Got an error when trying to install node where root could access. 
pi@raspberrypi:~ $ sudo ln -s /home/pi/.nvm/versions/node/v5.7.0/bin/node /usr/bin/node
ln: failed to create symbolic link ‘/usr/bin/node’: File exists

File: docs/getting-started/boards/getting-started-with-raspberry-pi.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+## Install Dependencies
+
```
Comment (cephirothdy2j) [inline docs/getting-started/boards/getting-started-with-raspberry-pi.md]: Are we expecting people to go through these steps or to just clone the example and npm install? If the former, do we need to include all the stuff about creating a directory and a package.json file and all that?

File: docs/getting-started/boards/getting-started-with-raspberry-pi.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+Next, add a [new device](/devices/overview) for the Raspberry PI. As mentioned before, this example only reports state when the button is pressed. So the only state attribute that's needed is `button`. Since a button usually just has an on/off value, set the data type to `Boolean`.
+
+![Create Raspberry Device](/images/getting-started/boards/create-raspberry-pi-device.png "Create Raspberry Device")
+
+Lastly, create an [access key](/applications/access-keys) that you'll use to authenticate the device's connection.
+
+## Send Button Presses
+
+Now we need the Raspberry PI to connect to Structure and begin sending state. Fortunately Johnny Five and the Structure SDK make this simple.
```
Comment (cephirothdy2j) [inline docs/getting-started/boards/getting-started-with-raspberry-pi.md]: Having not cloned the example repo and followed the steps, I'm finding it much easier to just ssh into the Pi. Should we call that out as an option?

File: docs/getting-started/boards/getting-started-with-raspberry-pi.md (hunk 1 above)
Comment (InconceivableDuck) [inline docs/getting-started/boards/getting-started-with-raspberry-pi.md]: It's a hard decision of where a tutorial stops. I think generally I was assuming the user knows how to setup the pi and knows they're going to use it to solve some problem. The starting point of this tutorial is basically then how to get it connected to Structure.

File: docs/getting-started/boards/getting-started-with-raspberry-pi.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+  ...
+  "dependencies": {
+    "johnny-five": "^0.9.26",
+    "raspi-io": "^5.3.0",
+    "structure-sdk-js": "^1.0.3"
+  }
+}
+```
+
+## Wiring
```
Comment (cephirothdy2j) [inline docs/getting-started/boards/getting-started-with-raspberry-pi.md]: I think this needs more explanation, similar to what we did for the hack night.

File: docs/getting-started/boards/getting-started-with-raspberry-pi.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+Now we need the Raspberry PI to connect to Structure and begin sending state. Fortunately Johnny Five and the Structure SDK make this simple.
+
+```JavaScript
+var five = require('johnny-five');
+var raspi = require('raspi-io');
+
+var Device = require('structure-sdk-js').Device;
+
+// Construct Structure device.
+var device = new Device({
```
Comment (cephirothdy2j) [inline docs/getting-started/boards/getting-started-with-raspberry-pi.md]: Forgot to switch these the first time I tried.

File: docs/getting-started/boards/getting-started-with-raspberry-pi.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+  // Hook the command event listener.
+  device.on('command', function(command) {
+    if(command.name === 'toggle') {
+      led.toggle();
+    }
+  });
+});
+```
+
```
Comment (cephirothdy2j) [inline docs/getting-started/boards/getting-started-with-raspberry-pi.md]: Note that we need to run as sudo?

File: docs/getting-started/boards/getting-started-with-raspberry-pi.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+The relevant code has been added to the board's ready callback. To listen for commands, the Structure device will emit the 'command' event whenever one is received from the platform. If the command's name is "toggle" the led is toggled.
+
+A typical way to send commands is by using a Structure workflow. Below is an example that will send the toggle command whenever a virtual button is pressed.
+
+![Command Workflow](/images/getting-started/boards/raspberry-commands.png "Command Workflow")
+
+Whenever the button on the above workflow is clicked, it will send a command with the name "toggle" to the connected Raspberry PI.
```
Comment (cephirothdy2j) [inline docs/getting-started/boards/getting-started-with-raspberry-pi.md]: I was always a fan of the "added challenge" that we'd include. Any way we can suggest something? e.g. build a workflow to tell the LED to flicker three times when the button is pressed?

**Pull request merged**